src/storage/store/compression.cpp

```diff
@@ -189,6 +220,36 @@ bool IntegerBitpacking<T>::canUpdateInPlace(T value, const BitpackHeader& header
     return true;
 }
 
+template<typename T>
+void fastunpack(const uint8_t* in, T* out, uint32_t bitWidth) {
+    if constexpr (std::is_same_v<std::make_signed_t<T>, int32_t> ||
+                  std::is_same_v<std::make_signed_t<T>, int64_t>) {
+        FastPForLib::fastunpack((const uint32_t*)in, out, bitWidth);
+    } else if constexpr (std::is_same_v<std::make_signed_t<T>, int16_t>) {
+        FastPForLib::fastunpack((const uint16_t*)in, out, bitWidth);
+    } else if constexpr (std::is_same_v<std::make_signed_t<T>, int8_t>) {
+        FastPForLib::fastunpack((const uint8_t*)in, out, bitWidth);
+    } else {
+        // Unreachable
+        assert(false);
+    }
+}
+
+template<typename T>
+void fastpack(const T* in, uint8_t* out, uint8_t bitWidth) {
+    if constexpr (std::is_same_v<std::make_signed_t<T>, int32_t> ||
+                  std::is_same_v<std::make_signed_t<T>, int64_t>) {
+        FastPForLib::fastpack(in, (uint32_t*)out, bitWidth);
+    } else if constexpr (std::is_same_v<std::make_signed_t<T>, int16_t>) {
+        FastPForLib::fastpack(in, (uint16_t*)out, bitWidth);
+    } else if constexpr (std::is_same_v<std::make_signed_t<T>, int8_t>) {
+        FastPForLib::fastpack(in, (uint8_t*)out, bitWidth);
+    } else {
+        // Unreachable
+        assert(false);
```

Comment:
Our convention is to not use `assert(false)`, instead `throw NotImplementedException("fastpack")`. mainly because of that in release mode, the `assert(false)` is just simply ignored and triggers compiler complaints.

Reply:
I've moved the last reachable branch into the else and added static_assert instead of its former condition so that we get a compile time failure in future if an unexpected type is used.

What I'd really wanted to use was `static_assert(false)`, but that doesn't work properly because it's not dependent on the template type.